src/battle_ai_switch_items.c

```diff
@@ -28,7 +28,43 @@ static bool8 ShouldUseItem(void);
 static bool32 AiExpectsToFaintPlayer(void);
 static bool32 AI_ShouldHeal(u32 healAmount);
 static bool32 AI_OpponentCanFaintAiWithMod(u32 healAmount);
-static bool32 IsAiPartyMonOHKOBy(u32 battlerAtk, struct Pokemon *aiMon);
+EWRAM_DATA struct SwitchinCandidate switchinCandidate = {0};
```

Comment:
This is the last change I need to make before it's ready for review, but this is fully unfamiliar to me and I can't get it to work properly. Writing it to EWRAM like this was just my copying how I found other structs being used in the codebase.

I tried mimicking the pattern of other structs in gBattleResources, both without a pointer and with a pointer and a define in the same style of AI_THINKING_STRUCT etc., and they compile, but they break AI switching after a KO.

Any advice?

Reply:
Did you try to put it in the AiLogicData struct?

The struct would look like this:
```
struct AiLogicData
{
    ...
    bool8 shouldSwitchMon; // Because all available moves have no/little effect. Each bit per battler.
    u8 monToSwitchId[MAX_BATTLERS_COUNT]; // ID of the mon to switch.
    struct SwitchinCandidate switchinCandidate;
};
```

Then, you would access it by either ```AI_DATA->switchinCandidate``` or ```gBattleResources->aiData->switchinCandidate``` 

Remember to use `make` and not `make NODEP=1` as the latter doesn't scan .h files.

Reply:
Did not realize I could just do that, whoops. I warned you I was new to structs in C lol

Worked without any issues, I'll get the PR updated shortly. Thanks again Egg! :D